### PR TITLE
Fix impersonated account updates

### PR DESCRIPTION
**Pull request by mdgreenfield**
# Overview

When updates come in, the account was being looked up and passed by reference, instead of copied, before applying updates. Without this change, users trying to update `token_scopes` will not be able to.

# Contributor Checklist
- [X] Add output for any tests not ran in CI to the PR description (eg, acceptance tests)
- [X] Backwards compatible

### CODE REVIEW

**Comment by mdgreenfield** [comment]
```text
➜  vault-plugin-secrets-gcp git:(mdgreenfield/impersonated-account-update-fixes) make testacc
?       github.com/hashicorp/vault-plugin-secrets-gcp   [no test files]
ok      github.com/hashicorp/vault-plugin-secrets-gcp/plugin    176.524s
?       github.com/hashicorp/vault-plugin-secrets-gcp/plugin/cache      [no test files]
ok      github.com/hashicorp/vault-plugin-secrets-gcp/plugin/iamutil    (cached)
?       github.com/hashicorp/vault-plugin-secrets-gcp/plugin/iamutil/internal   [no test files]
ok      github.com/hashicorp/vault-plugin-secrets-gcp/plugin/util       (cached)
?       github.com/hashicorp/vault-plugin-secrets-gcp/scripts/gohelpers [no test files]
```

**Review by robmonte — APPROVED** [review]
Looks good, worked as expected for me and tests passed. I believe @austingebauer wants to give it a review as well, so will wait to merge until he does so too.

File: plugin/path_impersonated_account_test.go [hunk 1]
```diff
@@ -96,6 +96,70 @@ func TestPathImpersonate_TTL(t *testing.T) {
 	testImpersonateDelete(t, td, impersonateName)
 }
 
+// Tests that fields can be updated
+func TestPathImpersonate_Update(t *testing.T) {
```
Comment (austingebauer) [inline plugin/path_impersonated_account_test.go:100]: Thanks for adding this!

**Review by austingebauer — APPROVED** [review]
LGTM. Thanks for catching this @mdgreenfield.

**Pull request merged**